runtime/compiler/control/JITClientCompilationThread.cpp

```diff
@@ -2950,18 +2950,19 @@ remoteCompilationEnd(
    }
 
 void
-updateCompThreadActivationPolicy(TR::CompilationInfoPerThreadBase *compInfoPT, JITServer::ServerMemoryState nextMemoryState)
+updateCompThreadActivationPolicy(TR::CompilationInfoPerThreadBase *compInfoPT, JITServer::ServerMemoryState nextMemoryState
+   JITServer::ServerActiveThreadState nextActiveThreadState)
    {
-   // When the server starts running low on memory, clients need to reduce load on the server.
-   // This is achieved by suspending compilation threads once low server memory is detected.
+   // When the server starts running low on memory or uses high number of compilation threads, clients need to reduce load on the server.
+   // This is achieved by suspending compilation threads once low server memory or high number of compilation threads are detected.
    //
    // Under normal conditions, a client uses AGGRESSIVE activation policy, starting new compilation
    // threads at lower thresholds then non-JITServer.
    //
-   // Once the server reaches LOW threshold, the client enters MAINTAIN activation policy,
+   // Once the server reaches LOW threshold for memory or HIGH threshold for active threads, the client enters MAINTAIN activation policy,
    // it keeps the current threads running but is not allowed to start new ones.
    //
-   // If the situation deteriorates further and the server reaches VERY_LOW threshold,
+   // If the situation deteriorates further and the server reaches VERY_LOW threshold for memory or very HIGH threshold for active threads,
    // client begins SUSPEND activation policy, disabling all but one compilation threads.
```

Comment:
Small typo in old comment: "but one compilation threads" --> "but one compilation thread"